src/gerbil/core/contract.ss

```diff
@@ -75,7 +75,23 @@ package: gerbil/core
     (and (identifier? stx)
          (alet (e (syntax-local-value stx false))
            (and (interface-info? e)
-                (is? e))))))
+                (is? e)))))
+
+  (def (interface-info-method-signature info method)
+    (let* ((info (cond ((interface-info? info) info)
+		       ((syntax-local-interface-info? info)
+			(syntax-local-value info false))
+		       (else
+			(raise-syntax-error #f "Unknown interface" info))))
+
+	   (method (cond ((symbol? method) method)
+			 ((identifier? method) (stx-e method))
+			 (else (raise-syntax-error #f "unknown interface method" info method))))
+	   (mdef (find (lambda (ms) (eq? method (car ms)))
+                       (interface-info-interface-methods info))))
+      (if mdef
+	(with ([_ sig _] mdef) sig)
+	(raise-syntax-error #f "unknown interface method" info method)))))
```

Comment:
it is more concise to write this as:
```
(match sig
 ([_ sig _] sig)
 (else (raise-syntax-error ...)))
```

Or maybe just prettier, but the point is lets try to use code that is a joy to read!
```